test/PerunPlutus/PerunDummySpec.hs

```diff
@@ -83,106 +108,159 @@ instance ContractModel PerunModel where
   -- channel.
   -- Furthermore the `Contract` managing the funds has to know that it holds
   -- the balances for each wallet.
-  nextState (Open wf (_, _) cid balA balB) = do
+  nextState (Start parties cid startBalances timeLock) = do
     modifyContractState $ \_ ->
       PerunModel
-        ( Just $
-            ChannelState
+        ( Just
+            (ChannelState
               { channelId = cid,
-                balanceA = balA,
-                balanceB = balB,
+                balances = startBalances,
                 version = 0,
                 final = False
-              }
+              }, timeLock, head startBalances : tail (map (const 0) startBalances), False)
+        )
+    withdraw (head parties) $ Ada.lovelaceValueOf $ head startBalances
+    wait 1
+  nextState (Fund funder index cid) = do
+    modifyContractState (\case
+        PerunModel Nothing -> P.error "Funding only works on existing channels"
+        PerunModel (Just (_, _, _, True)) -> P.error "Funding only works on unfunded channels"
```

Comment:
I do not think we want to assert this case in the state transition. This would forbid us to write _funding_ tests on a channel which should not be fundable.

Reply:
I would consider trying to fund a funded channel as malicious behaviour, as it is not a "valid" protocol step.
Therefore I would suggest to move this functionality into a `MaliciousFund` action (as we have discussed for `MaliciousDispute` etc.).
Do you agree with that?